### ISSUE TITLE
onClick (and other events) behaviors are not consistent

### ISSUE THREAD

**Issue by lazywei** [react-bootstrap/react-bootstrap#373]
The event `onClick`, and other events, behaviors are not consistent.
Some of them have behavior like React's synthetic event, i.e., they pass an `event` object as the first argument. However, some of them are not!

This is a little frustrating as I have to figure out the exact behavior for each one of them ...

Any idea on how to solve this?
Thanks.

**Comment by mtscout6** [comment]
I agree that the event param first approach should be our goal. Can you make a list of the offenders so we can tackle them?

**Comment by lazywei** [comment]
Do you have some plans or thoughts on how to design this in your mind? If not too hard, I'm thinking maybe I can help with this issue!

**Comment by mtscout6** [comment]
I'm a huge fan of consistency. Which particular components are you seeing that are not consistent? We can formulate a plan once we are aware of which components are part of the problem.

**Comment by lazywei** [comment]
OK, so for example, the `onClick` on `ListGroupItem` will not pass any argument, I have to do it myself:

```
<ListGroupItem key={key} href="" onClick={_this.handleOnClick.bind(null, otherUser.username)}> {otherUser.name} </ListGroupItem>
```

However, other components, e.g., Button, will pass react's synthetic event as the first argument, which I can access the `event.target` or `event.preventDefault()`.

**Comment by mtscout6** [comment]
I think that methods like the one in ListGroupItem

```
  handleClick: function (e) {
    if (this.props.onClick) {
      e.preventDefault();
      this.props.onClick(this.props.eventKey, this.props.href, this.props.target);
    }
  }
```

Should be

```
  handleClick: function (e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
```

Which actually has me wondering why we are handling that click at all. We can just pass the onClick prop along and drop this handleClick method all together.

Also, on a side note that has been on my mind lately is with regards to Accessibility concerns here. I can't tab to this item and control it with the keyboard, unless I add an href. Is an anchor tag the right option here?

**Comment by stefanfisk** [comment]
this just confused my for a good hour as well :(

**Comment by mtscout6** [comment]
Still confused? What can I elaborate on?

**Comment by stefanfisk** [comment]
Once I finally realized that the onClick was not being passed an event like I expected everything was clear, but I'm still stuck because I need access to the event to do what I want without writing a bunch of extra code :(

**Comment by stefanfisk** [comment]
so it seems like I'm still wrapping my head around how the click event is being handled...

since `preventDefault()` is called by the component  if there's a `onClick` handler one cannot conditionally do a `preventDefault()`, and it's also close to impossible to pass an event handler from an existing library since the argument signature is completely different and lacking the event.

What I want to do is really just use a `ListGroupItem` with `react-router` such that it only routes if click with the left mouse button and without any modifiers being pressed (to retain behaviours such as opening in a new tab), but I can't figure out how to do it with the current strangeness.

**Comment by mtscout6** [comment]
For that scenario it would make sense to add another component to [react-router-bootstrap](https://github.com/mtscout6/react-router-bootstrap) to wrap that as you expect. And fixing this bug would be a pre-requisite for that.

**Comment by stefanfisk** [comment]
Yep. For now I just wrapped the thing in a span and to which I then added my onClick handler, but I reckon that if this event handler signature is needed the property should be renamed to avoid confusion.

**Comment by stefanfisk** [comment]
Any update on this, do we have a go-ahead on removing the non-standard onClick handler?

**Comment by mtscout6** [comment]
Ya, let's go ahead and remove that handler here. Should be a quick pull request.

**Comment by mtscout6** [comment]
Relates to #419

**Comment by mtscout6** [comment]
@lazywei You fine with tracking progress on this in #419?

**Comment by lazywei** [comment]
sure, feel free to close this issue if you would like to :)
Thanks.

**Comment by ranjan653** [comment]
Hi onclick not working on react-bootstrap Tab

<Tab eventKey={2} title="Constraints" onClick={this.handleSelectTab}>
            <div>  </div>
            </Tab>

**Comment by androidand** [comment]
This is an old thread but I stumbled upon this inconsistency today. I am conditionally rendering a react-bootstrap ListItem or a custom component containing a react-bootstrap Card component. The ListItem works with onClick, the other does not. I ended up wrapping them both in a div and putting the onClick in it, worked for me:

<div onClick={props.onClick}>
	<CustomComponentThatMayOrMayNotHandleClickLikeIWant />
</div>
